packages/frontend/src/components/PositionCard.tsx

```diff
@@ -14,6 +14,7 @@ import { useTrade } from '@context/trade'
 import { useWorldContext } from '@context/world'
 import { PositionType, TradeType } from '../types'
 import { useVaultLiquidations } from '@hooks/contracts/useLiquidations'
+import { usePrevious } from 'react-use'
```

Comment:
what does this hook do / how is it working under the hood?

Reply:
It stores the previous value of a state. I realized that the issue was that the swaps data was set up to refetch 5secs after a successful trade but sometimes the refetch was not successful so it won't update the UI. So now I compare the previous swap state after a trade success to know if the state has been updated or not. 

Under the hood, it uses useRef hook for storing the previous state value. see [here](https://github.com/streamich/react-use/blob/master/src/usePrevious.ts)